From c895c51d84b36cf649764193f33f296728b27bdc Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 19 Apr 2008 17:43:42 +0200
Subject: [PATCH] __noreturn__ annotation for abort(3)

SUSv2 says about abort(3):

  The abort() function shall not return.

Hence, an '__attribute__((__noreturn__))' should be added to the
abort(3) declaration.
---
 include/stdlib.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/include/stdlib.h b/include/stdlib.h
index 34f3a7f..0cea604 100644
--- a/include/stdlib.h
+++ b/include/stdlib.h
@@ -47,7 +47,7 @@ double atof(const char *nptr) __THROW;
 __extension__ long long int atoll(const char *nptr);
 
 void exit(int status) __THROW __attribute__((__noreturn__));
-void abort(void) __THROW;
+void abort(void) __THROW __attribute__((__noreturn__));
 
 extern int rand(void) __THROW;
 extern int rand_r(unsigned int *seed) __THROW;
-- 
1.6.0.6

